### PR TITLE
Refactor our javascript autocompletion code

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
That's the first batch to get some initial feedback. I am currently checking how the more advanced autocompletes could be improved.

Once this got approved I am going to update our pattern library.

File: src/api/app/assets/javascripts/webui2/autocomplete.js [hunk 1]
```diff
@@ -1,29 +1,17 @@
...
-    source: sourcePath,
-    search: function(event, ui) { // jshint ignore:line
-      $(this).addClass('loading-spinner');
-    },
-    response: function(event, ui) { // jshint ignore:line
-      $(this).removeClass('loading-spinner');
-    },
-    minLength: 2});
-}
+$(document).ready(function() {
```
[the remark this answers is not in the document]
Comment (bgeuken) [reply src/api/app/assets/javascripts/webui2/autocomplete.js]: Sounds more like a complain about the asset pipeline. `$(document).ready()` just ensures that the document is loaded before being executed.

File: src/api/app/views/webui2/webui/project/_new_package_branch_modal.html.haml [hunk 2]
```diff
@@ -21,28 +21,12 @@
...
-            %label{ for: 'linked_package' } Orginal package name
-            .input-group
-              .input-group-prepend
-                %span.input-group-text#search-icon-package
-                  %i.fa.fa-search
-                  %i.fas.fa-spinner.fa-spin.d-none
-              = text_field_tag 'linked_package', nil, size: 80, class: 'form-control', placeholder: 'Type to autocomplete', required: true
+          = render partial: 'webui/autocomplete', locals: { html_id: 'linked_project', label: 'Orginal project name',
+                                                            source: autocomplete_projects_path }
+          = render partial: 'webui/autocomplete', locals: { html_id: 'linked_package', label: 'Orginal project name',
```
[the remark this answers is not in the document]
Comment (bgeuken) [reply src/api/app/views/webui2/webui/project/_new_package_branch_modal.html.haml]: @dmarcoux Nice catch. Fixed it now

**Comment by hennevogel** [comment]
There really is nothing ready made that can handle this?

**Comment by bgeuken** [comment]
Unfortunately not

**Pull request merged**